### PR TITLE
[feat] GUI Grundstruktur

### PR DESCRIPTION
**Pull request by joshuajeschek**
- [x] Icons setzten 
- [x] Analyse-Typ (MenuButton) -> ComboBox + aktuellen Typ im Feld
- [x] Bug: NullpointerException wenn kein Trial zu Beginn ausgewählt wurde

### CODE REVIEW

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package github.weichware10.analyse.gui;
+
+import github.weichware10.analyse.gui.util.AbsScene;
+import java.util.Optional;
+
+import javafx.scene.control.ButtonType;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'javafx.scene.control.ButtonType'

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import javafx.scene.control.Dialog;
+import javafx.scene.layout.BorderPane;
+
+public class TrialSelector extends AbsScene {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static BorderPane root;
+
+    public static String getTrialId() {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/analyse/gui/TrialSelectorController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import github.weichware10.analyse.gui.util.Log;
+import github.weichware10.util.Logger;
+import github.weichware10.util.ToolType;
+import javafx.fxml.FXML;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.DatePicker;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextField;
+
+public class TrialSelectorController extends AbsSceneController {
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelectorController.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.MissingJavadocTypeCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing a Javadoc comment.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import github.weichware10.analyse.gui.util.AbsScene;
+import github.weichware10.util.Logger;
+import github.weichware10.util.ToolType;
+import github.weichware10.util.data.TrialData;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+
+import javax.crypto.SecretKey;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'javax.crypto.SecretKey'

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+import javafx.application.Platform;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Extra separation in import group before 'javafx.application.Platform'

Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.application.Platform' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.event.ActionEvent;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.event.ActionEvent' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.fxml.FXMLLoader;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.fxml.FXMLLoader' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.scene.control.Button;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.control.Button' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.scene.control.ButtonType;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.control.ButtonType' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.scene.control.Dialog;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.control.Dialog' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.scene.control.ButtonBar.ButtonData;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.control.ButtonBar.ButtonData' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/TrialSelector.java [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+import javafx.scene.layout.BorderPane;
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/TrialSelector.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'javafx.scene.layout.BorderPane' import. Should be before 'javax.crypto.SecretKey'.

File: src/main/java/github/weichware10/analyse/gui/HeatmapConfigurator.java [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+ */
+public class HeatmapConfigurator extends AbsScene {
+
+    private static VBox root;
+    private static HeatmapConfiguratorController controller;
+
+    /**
+     * Startet die App.
+     *
+     * @return - Erstellte Konfiguration
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/HeatmapConfigurator.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused Javadoc tag.

File: src/main/java/github/weichware10/analyse/gui/HeatmapConfigurator.java [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public static void start(HeatmapConfig hmConfig) {
+        Dialog<String> configDialog = new Dialog<>();
+        ButtonType applyButtonType = new ButtonType("Übernehmen", ButtonData.APPLY);
+        configDialog.getDialogPane().getButtonTypes().addAll(applyButtonType, ButtonType.CANCEL);
+
+        configDialog.setTitle("Konfiguration Heatmap");
+
+        InitResult ir = initialize(HeatmapConfigurator.class.getResource("HeatmapConfigurator.fxml"));
```
Comment (weichbot10) [inline src/main/java/github/weichware10/analyse/gui/HeatmapConfigurator.java]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 102).

**Comment by Rotwolf** [comment]
Beim klicken auf "zurück" (in Analyse, Konfigurator. Trails erstellen) resized das Fenster auf eine kleiner Größe,
obwohl ich es vorher auf Vollbildschirm gezogen hatte. Ist das normal?

Außerdem werden im Konfigurator in dieser (automatisch verkleinerten) Größe beim ausklappen der Code Charts Konfiguration die unteren Button abgeschnitten (nicht dargestellt)

**Comment by joshuajeschek** [comment]
> Beim klicken auf "zurück" (in Analyse, Konfigurator. Trails erstellen) resized das Fenster auf eine kleiner Größe, obwohl ich es vorher auf Vollbildschirm gezogen hatte. Ist das normal?
> 
> Außerdem werden im Konfigurator in dieser (automatisch verkleinerten) Größe beim ausklappen der Code Charts Konfiguration die unteren Button abgeschnitten (nicht dargestellt)

ne, das ist nicht normal. da hat jmd nicht beachtet dass man an die start()-Funktion auch `null` als Größe übergeben kann

**Comment by joshuajeschek** [comment]
> Beim klicken auf "zurück" (in Analyse, Konfigurator. Trails erstellen) resized das Fenster auf eine kleiner Größe, obwohl ich es vorher auf Vollbildschirm gezogen hatte. Ist das normal?
> 
> Außerdem werden im Konfigurator in dieser (automatisch verkleinerten) Größe beim ausklappen der Code Charts Konfiguration die unteren Button abgeschnitten (nicht dargestellt)

Ich hab das jetzt mal so gemacht, dass sich die Fenstergröße nur ändert, wenn der Login-Bildschirm (Anfang) aufgerufen wird

**Review by Rotwolf — APPROVED** [review]

**Pull request merged**